betty/config/__init__.py

```diff
@@ -46,20 +47,32 @@ class Configurable(Generic[_ConfigurationT]):
     Any configurable object.
     """
 
-    _configuration: _ConfigurationT
+    def __init__(self, *args: Any, configuration: _ConfigurationT, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self._configuration = configuration
 
     @property
     def configuration(self) -> _ConfigurationT:
         """
         The object's configuration.
         """
-        if not hasattr(self, "_configuration"):
-            raise RuntimeError(
-                f"{self} has no configuration. {type(self)}.__init__() must ensure it is set."
-            )
         return self._configuration
 
 
+class DefaultConfigurable(Configurable[_ConfigurationT], Generic[_ConfigurationT]):
+    """
+    A configurable type that can provide its own default configuration.
+    """
+
+    @classmethod
+    @abstractmethod
+    async def default_configuration(cls) -> _ConfigurationT:
```

Comment:
Prefix with `new_` to communicate that each invocation must return a new instance.